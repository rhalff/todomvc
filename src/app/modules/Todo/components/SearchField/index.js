// @flow
import React from 'react'
import styles from './scss/SearchField.scss'

type Props = {
  checked: boolean,
  onChange: Function
}

const SearchField = ({ checked, onChange }: Props) => (
  <input
    className={checked ? styles.checked : styles.unchecked}
    type="checkbox"
    checked={checked}
    onChange={onChange}
  />
)

export default SearchField
