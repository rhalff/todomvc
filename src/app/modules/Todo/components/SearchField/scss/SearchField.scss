.searchField {
  position: absolute;
  top: -55px;
  right: -12px;
  width: 60px;
  height: 34px;
  text-align: center;
  border: none;
  background: none;
  transform: rotate(90deg);
  cursor: pointer;
}

.unchecked {
  composes: searchField;
}

.checked {
  composes: searchField;
}
