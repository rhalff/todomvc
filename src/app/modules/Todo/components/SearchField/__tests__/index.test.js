// @flow
import React from 'react'
import SearchField from '../'
import renderer from 'react-test-renderer'

test('Can show SearchField', () => {
  const component = renderer.create(<SearchField onChange={() => {}} />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Can show SearchField (checked)', () => {
  const component = renderer.create(<SearchField onChange={() => {}} checked />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
