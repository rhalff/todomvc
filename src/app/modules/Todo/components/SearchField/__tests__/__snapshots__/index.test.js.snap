// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show SearchField (checked) 1`] = `
<input
  checked={true}
  className="checked"
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`Can show SearchField 1`] = `
<input
  checked={undefined}
  className="unchecked"
  onChange={[Function]}
  type="checkbox"
/>
`;
